### PR TITLE
[omnibus] Make go mod cache configurable

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Makes the `GOMODCACHE` used in omnibus builds configurable, instead of a static location.
Fixes the Windows build scripts so that the `GOMODCACHE` used in `inv deps` and omnibus commands is the same.

### Motivation

In #7435, we switched to using `-mod=mod` when running `go` commands. In omnibus builds, we hardcoded the `GOMODCACHE` location to `/gomodcache`. This didn't break anything because the `GOMODCACHE` variable doesn't do anything until go 1.15: https://golang.org/doc/go1.15#tools, and we were still using go 1.14. Therefore omnibus builds were still using `$GOPATH/pkg/mod` as the module cache folder.

We moved to go 1.15 in MacOS builds a few days ago: https://github.com/DataDog/datadog-agent-macos-build/pull/15. With this change, the `GOMODCACHE` variable is now used, and go tries to write the module cache in `/gomodcache`. However, in the MacOS runners, we can't write in `/`, so this made MacOS Agent builds fail. Making this location configurable is therefore necessary.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]
Looks good, thanks for taking care of fixing this 🙇

File: tasks/winbuildscripts/dobuild.bat [hunk 1]
```diff
@@ -16,6 +16,7 @@ set OMNIBUS_ARGS=--python-runtimes "%PY_RUNTIMES%"
 if "%OMNIBUS_TARGET%" == "iot" set OMNIBUS_ARGS=--iot
 if "%OMNIBUS_TARGET%" == "dogstatsd" set OMNIBUS_BUILD=dogstatsd.omnibus-build && set OMNIBUS_ARGS=
 if "%OMNIBUS_TARGET%" == "agent_binaries" set OMNIBUS_ARGS=%OMNIBUS_ARGS% --agent-binaries
+if DEFINED GOMODCACHE set OMNIBUS_ARGS=%OMNIBUS_ARGS% --go-mod-cache %GOMODCACHE%
```
Comment (KSerrania) [inline tasks/winbuildscripts/dobuild.bat:19]: Technically not needed (given we also read the environment variable, but I prefer being explicit whenever possible.

**Review by olivielpeau — APPROVED** [review]
LGTM if CI passes!

Can you update the PR description to mention how this relates to https://github.com/DataDog/datadog-agent/pull/7435 and why this started breaking recently?

**Comment by KSerrania** [comment]
Description updated.

**Pull request merged**